src/common/error.ts

```diff
@@ -0,0 +1,47 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { print2 } from "../drivers/stdio";
+import Sentry from "./sentry";
+import { randomUUID } from "crypto";
+
+export const unknownErrorMessage = (errorId: string) =>
+  `P0 encountered an unknown error. Please contact support@p0.dev for assistance. (Error ID ${errorId})`;
+
+/** Implements a generic error boundary
+ *
+ * Logs an error to container logs and Sentry with a unique identifier.
+ *
+ * @returns A generic error message that should be served to the API caller.
+ */
+export const errorBoundary = (
+  error: any,
+  logContext: Record<string, any>,
+  debug?: boolean
+) => {
+  const errorId = randomUUID();
```

Comment:
I copied this logic from app. Sentry will generate an event id. And I am wondering why we generate our own?

Reply:
We didn't know probably, or, perhaps we didn't want to rely on Sentry if this event ID generation requires the Sentry API call to succeed.

Reply:
There's no need to generate this error ID IMO. In the P0 application, we do this to ensure that we don't leak any internal information to the user. But this is already running on the user's environment, and we won't have access to anything that the user already does not.